lib/cext/preprocess.rb

```diff
@@ -11,6 +11,26 @@
 
 ALLOCA_LOCALS = /^(\s+)VALUE\s*\*\s*([a-z_][a-zA-Z_0-9]*)\s*=\s*(\(\s*VALUE\s*\*\s*\)\s*)?alloca\(/
 
+# Found in nokogiri
+XML_NODE_SET_PATCH = <<-EOF
+  switch (rb_tr_to_int_const(rb_range_beg_len(arg, &beg, &len, (long)node_set->nodeNr, 0))) {
+  case Qfalse_int_const:
+    break;
+  case Qnil_int_const:
+EOF
+
+# Found in pg
+PG_BINARY_ENCODER_PATCH = <<-EOF	
+	switch(rb_tr_to_int_const(value)){
+		case Qtrue_int_const : mybool = 1; break;
+		case Qfalse_int_const : mybool = 0; break;
+EOF
+
+PATCHED_FILES = {'xml_node_set.c'      => {:line =>  "  switch (rb_range_beg_len(arg, &beg, &len, (long)node_set->nodeNr, 0)) {\n", 
+                                           :patch => XML_NODE_SET_PATCH},
+                 'pg_binary_encoder.c' => {:line => "	switch(value){\n",
+                                           :patch => PG_BINARY_ENCODER_PATCH}}
```

Comment:
@eregon It's currently matching the filename and line, Would adding line number to this work? That should be precise enough but may be too precise.  Ideas to add precision?

@chrisseaton The Puma/nio4r workarounds that are needed don't involve switching. I think it might be easier to handle on a case by case basis by the usages I've seen so far.